### PR TITLE
Show message when background audio recording is disabled

### LINKED ISSUE #4383: Recording overlay shows errors
**Acceptance**

- [x] Shows recording start errors just for background (pop up is also there)

- [x] Shows warning when recording disabled

### PR DESCRIPTION
**Pull request by seadowg**
Work towards #4383

#### What has been done to verify that this works as intended?

New tests and verified manually.

#### Why is this the best possible solution? Were any other approaches considered?

Comments inline.

#### How does this change affect users? Describe intentional changes to behavior and behavior that could have accidentally been affected by code changes. In other words, what are the regression risks?

Should just add the feature. Good to check background audio in it's entirety when this and everything else are merged to master.

#### Before submitting this PR, please make sure you have:
- [x] run `./gradlew checkAll` and confirmed all checks still pass OR confirm CircleCI build passes and run `./gradlew connectedDebugAndroidTest` locally.
- [x] verified that any code or assets from external sources are properly credited in comments and/or in the [about file](https://github.com/getodk/collect/blob/master/collect_app/src/main/assets/open_source_licenses.html).
- [x] verified that any new UI elements use theme colors. [UI Components Style guidelines](https://github.com/getodk/collect/blob/master/CONTRIBUTING.md#ui-components-style-guidelines)

### CODE REVIEW

File: collect_app/src/main/java/org/odk/collect/android/audio/AudioRecordingControllerFragment.java [hunk 1]
```diff
@@ -61,49 +64,64 @@ public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup c
...
-                }
-
-                if (backgroundAudioViewModel.isBackgroundRecording()) {
-                    binding.pauseRecording.setVisibility(GONE);
-                    binding.stopRecording.setVisibility(GONE);
-                }
-            } else {
-                binding.getRoot().setVisibility(GONE);
-            }
+        zip3(formEntryViewModel.hasBackgroundRecording(), backgroundAudioViewModel.isBackgroundRecordingEnabled(), audioRecorder.getCurrentSession()).observe(getViewLifecycleOwner(), triple -> {
```
Comment (seadowg) [inline collect_app/src/main/java/org/odk/collect/android/audio/AudioRecordingControllerFragment.java:67]: I looked at removing the zip here and seperating it out to different observations but I ended up in trouble as I had to make some of the same decisions in each observation (the `null` session case needs to reach out to `hasBackgroundRecording` vice versa).

I also tried using a "React-esque" approach of setting fields on the fragment for each observation and then updating, but I ended up in trouble as the first observation would fire before the next and so with unset fields being using to make decisions.

Comment (lognaturel) [reply collect_app/src/main/java/org/odk/collect/android/audio/AudioRecordingControllerFragment.java:67]: Thanks for looking into it. Let's go with this.

Comment (seadowg) [reply collect_app/src/main/java/org/odk/collect/android/audio/AudioRecordingControllerFragment.java:67]: Let's pair on this when we can as I think it's probably a nice test of how we're going to use `LiveData`.

File: collect_app/src/test/java/org/odk/collect/android/audio/AudioRecordingControllerFragmentTest.java [hunk 2]
```diff
@@ -193,4 +226,30 @@ public void whenThereIsAnErrorStartingRecording_showsErrorDialog() {
...
+            assertThat(fragment.binding.getRoot().getVisibility(), is(View.VISIBLE));
+            assertThat(fragment.binding.timeCode.getText(), is(fragment.getString(R.string.recording_disabled, "⋮")));
+            assertThat(fragment.binding.pauseRecording.getVisibility(), is(View.GONE));
+            assertThat(fragment.binding.stopRecording.getVisibility(), is(View.GONE));
+            assertThat(fragment.binding.waveform.getVisibility(), is(View.GONE));
+        });
+    }
+
+    @Test
+    public void whenFormDoesNotHaveBackgroundRecording_andBackgroundRecordingIsDisabled_showsThatRecordingIsDisabled() {
```
Comment (lognaturel) [inline collect_app/src/test/java/org/odk/collect/android/audio/AudioRecordingControllerFragmentTest.java]: Is this really `doesNotShowThatRecordingIsDisabled`?

File: strings/src/main/res/values/strings.xml [hunk 3]
```diff
@@ -196,7 +196,12 @@
 
     <!-- Error message displayed when copying media from a temporary directory fails -->
     <string name="answer_file_copy_failed_message">Media could not be attached to form but is available at %1$s</string>
-    
+
+    <string name="saving_audio_recording_failed">Saving audio recording failed</string>
```
Comment (lognaturel) [inline strings/src/main/res/values/strings.xml]: I don't think this belongs here.

**Review by lognaturel — APPROVED** [review]

File: strings/src/main/res/values/strings.xml (hunk 3 above)
Comment (seadowg) [reply strings/src/main/res/values/strings.xml]: It felt related to the answer file copy failure to me. Where do you think it should go?

Comment (lognaturel) [reply strings/src/main/res/values/strings.xml]: Isn't it a new string that's not actually used?!

Comment (seadowg) [reply strings/src/main/res/values/strings.xml]: huh I see! Hmmm I'll fix, but I'm weirded out that lint didn't catch this...

**Comment by mmarciniak90** [comment]
**Tested with success**

Verified on Android versions: 7.0, 8.1, 10.0

| Screenshot_20210211-085113|Screenshot_20210211-090912|
|---|---|

Verified cases:
- clear message is visible when recording is disabled
- different font size
- new form with disabled recording
- edit saved form
- light/dark mode
- English/Spanish language
- RTL language
- minimize
- rotation

**Comment by kkrawczyk123** [comment]
Verified also on Androids: 5.1, 7.1 and 9.0.

@getodk-bot unlabel "needs testing"
@getodk-bot label "behavior verified"

**Pull request merged**